src/components/folder/FolderList.js

```diff
@@ -1,59 +1,188 @@
-import { getAllFolderLink } from '../../api';
-import { useEffect, useState } from 'react';
-import Card from '../shared/Card';
-import '../styles/card.css';
+import share from '../../images/share.svg';
+import pen from '../../images/pen.svg';
+import deleteicon from '../../images/delete.svg';
+
+import { getAllFolderLink, getFolderLinkData } from '../../api';
+import { React, useEffect, useState } from 'react';
+
+import styled from 'styled-components';
+
+const MainContainer = styled.div`
+  display: flex;
+  column-gap: 2rem;
+  row-gap: 2.5rem;
+  justify-content: space-between;
+  flex-wrap: wrap;
+  width: 106rem;
+  margin: 0 auto;
+
+  @media (max-width: 1123px) {
+    column-gap: 2.4rem;
+    padding: 0 3.2rem;
+    width: 76.8rem;
+  }
+
+  @media (max-width: 767px) {
+    row-gap: 2rem;
+    width: 38.9rem;
+  }
+`;
+
+const MainContainerEmpty = styled.div`
+  color: #000;
+  text-align: center;
+  font-size: 1.6rem;
+  font-weight: 400;
+  line-height: 2.4rem;
+  padding: 4rem;
+`;
+
+const Folder = styled.div`
+  align-items: center;
+  display: grid;
+  grid-template-areas: 'folders folders add-button';
+  justify-content: space-between;
+`;
+
+const FolderContainer = styled.div`
+  width: 100%;
+`;
+
+const OptionContainer = styled.div`
+  display: flex;
+  width: 1060px;
+  justify-content: space-between;
+  align-items: center;
+  margin-top: 24px;
+`;
+
+const Options = styled.div`
+  display: flex;
+  align-items: center;
+  gap: 1.2rem;
+`;
+
+const FolderName = styled.h3`
+  color: #000;
+  font-family: Pretendard;
+  font-size: 24px;
+  font-weight: 600;
+  letter-spacing: -0.2px;
+`;
+
+const FolderAddButton = styled.a`
+  color: #6d6afe;
+  font-family: Pretendard;
+  font-size: 16px;
+  font-style: normal;
+  font-weight: 500;
+  line-height: normal;
+  letter-spacing: -0.3px;
+  display: flex;
+  gap: 4px;
+  justify-content: center;
+  align-items: center;
+  grid-area: add-button;
+`;
+
+const FolderTab = styled.div`
+  display: flex;
+  gap: 1.2rem;
+  flex-wrap: wrap;
+  grid-area: folders;
+  column-gap: 0.8rem;
+`;
+
+const OptionIcon = styled.img`
+  width: 18px;
+  height: 18px;
+`;
+
+const Button = styled.button`
+  color: var(--gray60);
+  font-family: Pretendard;
+  font-size: 1.4rem;
+  font-weight: 600;
+
+  display: flex;
+  align-items: center;
+  gap: 4px;
+`;
 
 const FolderList = ({ folderList }) => {
-  const [folderLink, setFolderLink] = useState(null);
+  const [links, setLinks] = useState(null);
+  const [item, setItem] = useState({ id: 'all', name: '전체' });
+
+  const handleLoadAllFolderLink = async () => {
+    const { data, error } = await getAllFolderLink();
+    if (error) {
+      return alert(error);
+    }
+    setLinks(data);
+    setItem({ id: 'all', name: '전체' });
+  };
 
   const handleLoadFolderLink = async () => {
-    const links = await getAllFolderLink();
+    const { data, error } = await getFolderLinkData(item.id === 'all' ? '' : `${item.id}`);
```

Comment:
@JeongSabibi 

이것도 개인적인 선호사항인데요, 
필터 값을 연산하는 로직을 삼항으로 함수 호출부분 안에서 작성하기보다, 분리해서 관리해보면 어떨까요? 

```suggestion
    const filterId = item.id === "all" ? "" : item.id
    const { data, error } = await getFolderLinkData(filterId);
```

조금 더 나아간다면, 폴더 필터 구현 시, ALL의 경우를 그냥 null로 두고 하는게 조금 더 real world practice에 적합할 것 같아요.
실제로도 필터가 없다면 모든 데이터를 불러온다라고 할 수 있잖아요. 

따라서 
```jsx
const [folderFilter, setFolderFilter] = useState(null)

function handleFolderFilterSelect(filter) {
  setFolderFilter(filter)
}

async function handleLoadFolderLink() {
  const filterId = folderFilter ?? folderFilter.id  // null이라면 null 할당, null이 아니라면 필터id 할당
  const {data, error} = await getFolderLinkData(filterId)
}

export async function getFolderLinkData(folderId) {
  const queryParams = `?folderId=${folderId ? folderId : "all"}`
  const response = await fetch(`${BASE_URL}/users/${SAMPLE_ID}/links${queryParams}`);
...
}

```